### PR TITLE
SPU2: Purge interpolation options

### PR DESCRIPTION
**Pull request by JordanTheToaster**
### Description of Changes
Removes the options menu for interpolation as all but gaussian will be removed.

### Rationale behind Changes
Having a selection box for just one option is a bit silly I'll see if I can remove the actual code for it later on in another PR as that part frightens me more than this.

### Suggested Testing Steps
Make sure the CI doesn't explode and that audio still works.

### CODE REVIEW

File: pcsx2/SPU2/Mixer.cpp [hunk 1]
```diff
@@ -594,21 +594,6 @@ static __forceinline StereoOut32 MixVoice(uint coreidx, uint voiceidx)
...
-				case Pcsx2Config::SPU2Options::InterpolationMode::Cubic:
-					Value = GetVoiceValues<2>(thiscore, voiceidx);
-					break;
-				case Pcsx2Config::SPU2Options::InterpolationMode::Hermite:
-					Value = GetVoiceValues<3>(thiscore, voiceidx);
-					break;
-				case Pcsx2Config::SPU2Options::InterpolationMode::CatmullRom:
-					Value = GetVoiceValues<4>(thiscore, voiceidx);
-					break;
 				case Pcsx2Config::SPU2Options::InterpolationMode::Gaussian:
```
Comment (lightningterror) [inline pcsx2/SPU2/Mixer.cpp:597]: No need for the switch case anymore, code can be changed, maybe also remove the config itself for interpolation as there's no need.

Comment (JordanTheToaster) [reply pcsx2/SPU2/Mixer.cpp:597]: I'll mark as draft for now but I will remove the unnecessary bits as well.

File: pcsx2/Config.h [hunk 2]
```diff
@@ -780,11 +780,6 @@ struct Pcsx2Config
 	{
 		enum class InterpolationMode
```
Comment (lightningterror) [inline pcsx2/Config.h:781]: Can remove the entire enum.

File: pcsx2/SPU2/Mixer.cpp [hunk 3]
```diff
@@ -585,37 +585,8 @@ static __forceinline StereoOut32 MixVoice(uint coreidx, uint voiceidx)
...
-					break;
-				case Pcsx2Config::SPU2Options::InterpolationMode::Gaussian:
-					Value = GetVoiceValues<5>(thiscore, voiceidx);
-					break;
-
-					jNO_DEFAULT;
-			}
-		}
+	
+		Value = GetVoiceValues<5>(thiscore, voiceidx);
```
Comment (refractionpcsx2) [inline pcsx2/SPU2/Mixer.cpp:589]: can you restore noise handling please?

Thanks

Comment (refractionpcsx2) [reply pcsx2/SPU2/Mixer.cpp:589]: probably should update GetVoiceValues too to not take a template number and just default to interpolating

File: pcsx2/Config.h [hunk 4]
```diff
@@ -778,15 +778,6 @@ struct Pcsx2Config
...
 	{
-		enum class InterpolationMode
-		{
-			Nearest,
-			Linear,
-			Cubic,
-			Hermite,
-			CatmullRom,
-			Gaussian
-		};
```
Comment (refractionpcsx2) [inline pcsx2/Config.h:781]: line to get rid of

**Review by refractionpcsx2 — APPROVED** [review]

**Comment by Ziemas** [comment]
What about removing the now unused interpolation functions?

**Comment by refractionpcsx2** [comment]
> What about removing the now unused interpolation functions?

fair point :P I forgot about those...

**Comment by JordanTheToaster** [comment]
> What about removing the now unused interpolation functions?

Done.

File: pcsx2/SPU2/Mixer.cpp [hunk 5]
```diff
@@ -373,120 +373,25 @@ __forceinline static s32 GaussianInterpolate(s32 pv4, s32 pv3, s32 pv2, s32 pv1,
...
-	s32 val = ((a0)*mu) >> 12;
-	val = ((val + a1) * mu) >> 12;
-	val = ((val + a2) * mu) >> 12;
-
-	return (val + y1);
-}
-
 // Returns a 16 bit result in Value.
 // Uses standard template-style optimization techniques to statically generate five different
 // versions of this function (one for each type of interpolation).
```
Comment (Ziemas) [inline pcsx2/SPU2/Mixer.cpp:378]: Can delete these two comment lines as they no longer apply.

Comment (JordanTheToaster) [reply pcsx2/SPU2/Mixer.cpp]: I may be blind.

**Review by Mrlinkwii — APPROVED** [review]

**Pull request merged**